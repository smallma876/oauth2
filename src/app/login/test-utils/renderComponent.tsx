import { act, render, RenderResult } from '@testing-library/react';
import { ReactElement } from 'react';
import { GlobalAppProvider } from '../../../store/app-context';
import { AppState, DispatchApp } from '../../../store/reducer';
import { LoginState } from '../store/reducer';
import * as context from './../store/login-context';

interface RenderComponentInterface {
  Component: ReactElement;
  stateProps?: Partial<AppState>;
  extraState?: Partial<LoginState>;
  dispatch?: DispatchApp;
}

export const renderComponent = async (renderComponentConfig: RenderComponentInterface) => {
  let component = {} as RenderResult;
  const { stateProps, dispatch, Component, extraState = {} } = renderComponentConfig;

  await act(async () => {
    component = render(
      <GlobalAppProvider testStateProps={stateProps} testDispatch={dispatch}>
        <context.LoginProvider extraState={extraState}>{Component}</context.LoginProvider>
      </GlobalAppProvider>,
    );
  });

  return component;
};
