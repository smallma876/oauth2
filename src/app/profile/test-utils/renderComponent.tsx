import { act, render, RenderResult } from '@testing-library/react';
import { ReactElement } from 'react';
import { GlobalAppProvider } from '../../../store/app-context';
import { AppState, DispatchApp } from '../../../store/reducer';
import * as context from '../store/profile-context';
import { ProfileState } from '../store/reducer';

interface RenderComponentInterface {
  Component: ReactElement;
  stateProps?: Partial<AppState>;
  extraState?: Partial<ProfileState>;
  dispatch?: DispatchApp;
}

export const renderComponent = async (renderComponentConfig: RenderComponentInterface) => {
  let component = {} as RenderResult;
  const { stateProps, dispatch, Component, extraState = {} } = renderComponentConfig;

  await act(async () => {
    component = render(
      <GlobalAppProvider testStateProps={stateProps} testDispatch={dispatch}>
        <context.ProfileProvider extraState={extraState}>{Component}</context.ProfileProvider>
      </GlobalAppProvider>,
    );
  });

  return component;
};
