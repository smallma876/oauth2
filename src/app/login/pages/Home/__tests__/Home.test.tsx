import { MemoryRouter } from 'react-router-dom';
import Login from '../../../Login';
import { renderComponent } from '../../../test-utils/renderComponent';

describe('home test', () => {
  it('render home', async () => {
    const { container } = await renderComponent({
      Component: (
        <MemoryRouter initialEntries={['/']}>
          <Login />
        </MemoryRouter>
      ),
    });
    expect(container).toMatchSnapshot();
  });
});
