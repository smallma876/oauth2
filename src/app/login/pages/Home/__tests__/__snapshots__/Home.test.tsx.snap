// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home test render home 1`] = `
<div>
  <div>
    <div>
      OAUTH 2.0
    </div>
    <div>
      Login with: 
    </div>
    <div
      id="signInDiv"
    />
  </div>
</div>
`;
